Add dedicated accordion component

mdo opened the pull request:
This PR replaces and adds onto #30042. The goal of this is to replace our half-baked card accordion with a fully functioning accordion component, powered by the Collapse plugin. While this adds some CSS, personally I think it makes accordions more widely accessible (e.g., fewer requirements to creating the component).

According to the origin PR by @gijsroge, this does the following:

- Fixes #30015, the original feature request issues
- Fixes #28134 by moving the border-bottom from the header to a border-top on the body
- Fixes #28873 by removing the `overflow: hidden;` on the .accordion-item
- Fixes #25811 by adding an `.accordion-flush` variant (in addition to `.accordion-striped`, which may not be necessary IMO)

Also from that PR, there were some remaining todos. Design and code are largely final now with my latest changes here. Looking through https://www.w3.org/TR/wai-aria-practices-1.1/#accordion, I think we're largely in order. There's also this handy dandy [example accordion](https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html) they provide.

**Preview:** https://deploy-preview-32013--twbs-bootstrap.netlify.app/docs/5.0/components/accordion/

---

Some open questions from me now moving this forward:

- [x] Do we need striping here? I think not, and may remove it shortly.
- [x] Any other variations, documentation, etc we should be considering?

/cc @twbs/css-review

Hiws commented on scss/_card.scss:
```diff
@@ -213,30 +213,3 @@
-//
-
-.accordion {
-  overflow-anchor: none;
```
I think `overflow-anchor: none;` should be added to `.accordion-button` to avoid reintroducing the issue fixed in https://github.com/twbs/bootstrap/pull/31346

XhmikosR commented:
@mdo this needs a proper rebase, preferably squashing the original PR's patches into one.

mdo committed:
Add dedicated accordion component based on Collapse JS

MartijnCuppens commented on scss/_accordion.scss:
```diff
@@ -0,0 +1,124 @@
+    background-image: escape-svg($accordion-button-icon);
+    background-repeat: no-repeat;
+    background-size: $accordion-icon-width;
+    transform-origin: center center;
```
This is the default transform-origin, or is there any other reason why this is here?
```suggestion
```

MartijnCuppens commented on scss/_accordion.scss:
```diff
@@ -0,0 +1,124 @@
+    content: "";
+    background-image: escape-svg($accordion-button-icon);
+    background-repeat: no-repeat;
+    background-size: $accordion-icon-width;
+    transform-origin: center center;
```
This is the default transform-origin, or is there any other reason why this is here?
```suggestion
    background-size: $accordion-icon-width;
```

mdo replied:
Leftover from me. I was debugging the original PR's problem of a flashing background resize. Was playing with ideas before I realized the transition was set to `all` and the `background-image` was being swapped mid-transition.

mdo replied:
Strange, I can't accept your suggested change. See anything weird on your end? 🤔

MartijnCuppens replied:
Hmm, nope. I think github just can't handle removing lines in comments.

MartijnCuppens commented on scss/_accordion.scss:
```diff
@@ -0,0 +1,124 @@
+  }
+
+  &:focus {
+    position: relative;
```
What's the purpose of this rule?

mdo replied:
This effectively brings the focused accordion button to the top of the stack—it's like our `z-index: 1` tricks elsewhere I think, minus the `z-index`. Thought it was a little clever, so unsure if we should add the `position` by default and toggle `z-index` instead. Thoughts?

MartijnCuppens replied:
I'm ok with this shorter solution, but we might need to explain its purpose in the source code, because it's not self explanatory.

MartijnCuppens commented on scss/_variables.scss:
```diff
@@ -1014,6 +1014,33 @@ $card-img-overlay-padding:          $spacer !default;
+$accordion-button-color:                  $accordion-color !default;
+$accordion-button-bg:                     $accordion-bg !default;
+$accordion-button-active-bg:              tint-color($component-active-bg, 90%) !default;
+$accordion-button-active-color:           $primary !default;
```
Color contrast between this color and the background is less than 4.5. We might need to shade this color a bit?

MartijnCuppens commented:
The bottom border is missing when the last element is collapsing
![image](https://user-images.githubusercontent.com/11559216/97972406-c290d800-1dc4-11eb-8cd2-2571ff355a9d.png)

The bottom border is below the focus state here:
![image](https://user-images.githubusercontent.com/11559216/97972735-434fd400-1dc5-11eb-8973-8d3181b4787d.png)

patrickhlauke commented:
Just gave this a quick check in terms of accessibility - of course, as it's based on our current one, it's all kosher.

One small request though: would it be possible to ditch the latin placeholder text for some generic english text? Even if it's just something that is self explanatory, like "This is the example content for the accordion panel. It is revealed when the accordion control is activated." or similar? It's not a huge deal, but it avoids having the page all of a sudden switch to absolutely incomprehensible gibberish for screen reader users, for instance.

mdo committed:
Update relative position hack for layers

Switches to slightly more verbose, but more consistent, z-index layering we use elsewhere (e.g., pagination). Doing this ensures we're not toggling position on and off, but rather z-index.

mdo committed:
Remove default transform-origin per code review

MartijnCuppens committed:
Fix missing border & add transitions

MartijnCuppens committed:
Shade color for better color contrast

MartijnCuppens commented:
Fixed the remaining bugs by adding an additional `accordion-collapse` class.

mdo commented:
> Fixed the remaining bugs by adding an additional `accordion-collapse` class.

I... think I love you? Thank you for the fixes!

XhmikosR committed:
_variables.scss: reorder SVG attributes

This results in slightly better gzip compression

XhmikosR commented:
Saved a few more bytes by reordering the SVG attributes :)

patrickhlauke commented:
> Fixed the remaining bugs by adding an additional `accordion-collapse` class.

oh, i just noticed that includes also changing the latin text. thank you, top stuff! (funnily enough, I've just written up a bug for another project that also uses lorem ipsum for their demos :) )
